src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -391,6 +401,75 @@ protected virtual void Initialize()
             }
         }
 
+        // Scan the extensions directory and Load custom extension.
+        // $$$ - need to finalize this policy. Currently enabled via an AppSettting. 
+        private void LoadCustomExtensions(JObject hostConfig)
+        {
+            var bindingRoot = ConfigurationManager.AppSettings["BYOB_Path"];
+            if (!string.IsNullOrWhiteSpace(bindingRoot))
+            {
+
+                var config = this.ScriptConfig.HostConfig;
+
+                foreach (var dir in Directory.EnumerateDirectories(bindingRoot))
+                {
+                    foreach (var path in Directory.EnumerateFiles(dir, "*.dll"))
+                    {
+                        // How to find the extension assembly? Don't want to load every one to inspect it. 
+                        // $$$ manifest file? scan for any assembly with 
+                        if (!path.ToLowerInvariant().Contains("extension"))
+                        {
+                            continue;
+                        }
+
+                        // See GetNugetPackagesPath() for details 
+                        // Script runtime is already setup with assembly resolution hooks, so use LoadFrom
+                        Assembly assembly = Assembly.LoadFrom(path);
```

Comment:
It would be VERY nice to load only the primary extension assembly (by addressing the comment above, that might help) instead of calling `LoadFrom` right away. We may want to consider a marker attribute and perform a reflection only load to check for its presence. That would actually help with the assembly load context issues you ran into.